pkg/jobs/hive/hive.go

```diff
@@ -329,8 +333,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
+		},
+		Spec: managedclusteractionv1beta1.ActionSpec{
+			ActionType: managedclusteractionv1beta1.UpdateActionType,
+			KubeWork: &managedclusteractionv1beta1.KubeWorkSpec{
+				Resource:       "configmaps",
+				Name:           "admin-acks",
+				Namespace:      "openshift-config",
+				ObjectTemplate: updateConfigMap,
+			},
+		},
+	}
+	if err := client.Create(context.TODO(), ocpConfigMCA); err != nil {
+		return err
+	}
+
+	// wait for managedclusteraction results
+	ocpConfigMCAStatus := managedclusteractionv1beta1.ManagedClusterAction{}
+	for i := 1; i <= 5; i++ {
+		time.Sleep(utils.PauseFiveSeconds)
+		if err := client.Get(context.TODO(), types.NamespacedName{
+			Name:      clusterName + "admack",
+			Namespace: clusterName,
+		}, &ocpConfigMCAStatus); err != nil {
+			if i == 5 {
+				return err
+			}
+			klog.Warning(err)
+			continue
+		}
+		if ocpConfigMCAStatus.Status.Conditions != nil {
+			break
+		}
+	}
+
+	if ocpConfigMCAStatus.Status.Conditions != nil {
+		for _, condition := range ocpConfigMCAStatus.Status.Conditions {
+			if condition.Status == v1.ConditionFalse && condition.Reason == managedclusteractionv1beta1.ReasonUpdateResourceFailed {
+				klog.Warning("ManagedClusterAction failed to update remote clusterversion", ocpConfigMCAStatus.Status.Conditions)
+				return errors.New("Remote confimap update failed")
+			}
+			if condition.Status == v1.ConditionTrue && condition.Type == managedclusteractionv1beta1.ConditionActionCompleted {
+				klog.V(2).Info("Remote configmap updated successfully " + clusterName + "admack")
+			}
+		}
+	} else {
+		return errors.New("Remote configmap update failed")
+	}
+
+	if err := client.Delete(context.TODO(), &ocpConfigMCAStatus); err != nil {
+		return err
+	}
+
+	// After updating the OCP ack configmap it takes some time for clusterversion to pick up the change
+	klog.V(0).Info("Pause 60 seconds for clusterversion to update")
+	time.Sleep(utils.PauseSixtySeconds)
+
+	var retries = 1
+	curatorAnnotations := curator.GetAnnotations()
+
+	if curatorAnnotations != nil && curatorAnnotations[UpgradeClusterversionBackoffLimit] != "" {
+		backoffLimit, err := strconv.Atoi(curatorAnnotations[UpgradeClusterversionBackoffLimit])
+		if err == nil {
+			if backoffLimit > 0 && backoffLimit <= 100 {
+				retries = backoffLimit
+			} else if backoffLimit > 100 {
+				retries = 100
+			}
+		}
+	}
+
+	klog.V(0).Info("Retries set to: " + strconv.Itoa(retries))
+
+	successful := false
+	for i := 1; i <= retries && !successful; i++ {
+		klog.V(2).Info("Update clusterversion attempt " + strconv.Itoa(i))
+
+		mcaStatus, err := eusRetreiveAndUpdateClusterVersion(client,
+			clusterName, updateVersion, managedclusterview, isInterVersion)
+		if err != nil {
+			return err
+		}
+
+		if mcaStatus.Status.Conditions != nil {
+			for _, condition := range mcaStatus.Status.Conditions {
+				if condition.Status == v1.ConditionFalse && condition.Reason == managedclusteractionv1beta1.ReasonUpdateResourceFailed {
```

Comment:
Is this if check copy from somewhere else? Usually we find the condition using the type and not the reason.
There are also some nilpointer concerns.

I am thinking this block starting from `if mca.Status.Conditions != nil {` to the successful break should be:

```
if mca != nil {
  condition := meta.FindStatusCondition(mca.Status.Conditions, managedclusteractionv1beta1.ConditionActionCompleted)
  if condition != nil {
    if condition.Status == v1.ConditionTrue {
      ...break...
    } else if condition.Status == v1.ConditionFalse {
     ...retry
    }
  }
```

Reply:
Yeah this is copied from here https://github.com/stolostron/cluster-curator-controller/blob/main/pkg/jobs/hive/hive.go#L303

Let me update it.

Reply:
BTW if you look at the function eusRetreiveAndUpdateClusterVersion()
https://github.com/stolostron/cluster-curator-controller/pull/212/files#diff-2853d09c9e46ab43286b8a391d591993dd1783dea5f5135ecd9b3fb7c37825f7L494

it always returns an empty managedclusteractionv1beta1.ManagedClusterAction{} so nil pointer is not possible. Also it won't let me check for nil:
![image](https://github.com/stolostron/cluster-curator-controller/assets/38960034/7d3c0928-eb55-4038-b60a-b543f012b6c2)

So I'll just change how to find the condition using the type.